src/gfx/reverse.cpp

```diff
@@ -381,15 +443,14 @@ void reverse() {
 		for (size_t tx = 0; tx < width; ++tx) {
 			size_t index = options.columnMajor ? ty + tx * height : ty * width + tx;
 			// By default, a tile is unflipped, in bank 0, and uses palette #0
-			uint8_t attribute = attrmap.has_value() ? (*attrmap)[index] : 0x00;
+			uint8_t attribute = attrmap ? (*attrmap)[index] : 0x00;
 			bool bank = attribute & 0x08;
 			// Get the tile ID at this location
-			size_t tileID = index;
-			if (tilemap.has_value()) {
-				tileID =
-				    (*tilemap)[index] - options.baseTileIDs[bank] + bank * options.maxNbTiles[0];
-			}
-			assume(tileID < nbTileInstances); // Should have been checked earlier
+			size_t tileOfs =
+			    tilemap ? static_cast<uint8_t>((*tilemap)[index] - options.baseTileIDs[bank])
+			                  + (bank ? nbTilesInBank[0] : 0)
+			            : index;
+			assume(tileOfs < nbTiles + options.trim); // Should have been checked earlier
```

Comment:
Then maybe
```suggestion
			// Should have been enforced above, this assertion is here for what slips through the cracks.
			assume(tileOfs < nbTiles + options.trim);
```

Reply:
Or just "This should have been enforced above."  No need to explain that assertions are there to double-check preconditions just in case. (If we actually expected anything to "slip through the cracks", we should be fixing the code to not do that.)